flink-runtime/src/main/java/org/apache/flink/runtime/minicluster/MiniCluster.java

```diff
@@ -1179,11 +1193,16 @@ public void onFatalError(Throwable exception) {
     }
 
     private class ShutDownFatalErrorHandler implements FatalErrorHandler {
+        private boolean cleanupHaData;
+
+        ShutDownFatalErrorHandler(boolean cleanupHaData) {
+            this.cleanupHaData = cleanupHaData;
+        }
 
         @Override
         public void onFatalError(Throwable exception) {
             LOG.warn("Error in MiniCluster. Shutting the MiniCluster down.", exception);
-            closeAsync();
+            shutdownAsync(cleanupHaData);
```

Comment:
Not directly related to this issue but how is this shutdown on fatal error propagated? Will we exit the process? Do we restart the TaskManager process?